Add support for PHP 7.4

mmachatschek opened the pull request:
## Purpose

Change travis configuration and add support for PHP 7.4

#### Open Questions and Pre-Merge TODOs

- [x] `composer lint` and `composer fix` was executed.
- [x] Tests were written and pass with 100% coverage.
- [ ] ~~A issue with a detailed explanation of the problem/enhancement was created and linked.~~

Commit:
Add support for PHP 7.4

Commit:
Fix code coverage issue

mmachatschek commented:
@howard What do you think about dropping support for 1.x branch?

howard commented:
@mmachatschek 1.x is still in active use for some shop system plugins that have a very old PHP version baseline, so it's not time to do that yet.

Please limit the changes to what's necessary to make libflexport (2.x) *compatible* with PHP 7.4, without traces of introducing a new major version. That's a new topic that will soon get its own milestone issue to collect ideas and outline the plan for libflexport 3.0.

howard commented:
@mmachatschek Your input for libflexport 3.0 is welcome in #114.

Commit:
Add support for php 7.1 again

Commit:
Update readme

mmachatschek commented:
@howard I updated the code again. I added support for PHP 7.1, as we would need to release a major version if we drop PHP 7.1? If not, please tell me. I would gladly update the code again.

howard reviewed (approved):
@mmachatschek PHP 7.1 is still part of our compatibility matrix, and based on known PHP versions used by customers, it will remain supported, for 2.x at least.
